src/app/plugins/platform/Docker/dockerfile/createPHPDockerfile.ts

```diff
@@ -0,0 +1,163 @@
+import { Dockerfile, From } from 'dockerfilejs';
+
+import Dependency from './Dependency';
+
+/**
+ * Options needed to create a `Dockerfile` object for a PHP-based image.
+ */
+export interface CreatePHPDockerfileOptions {
+  /**
+   * The base image to create this from.
+   */
+  from: From;
+
+  /**
+   * Any dependencies (such as the gd builtin or memcached from PECL)
+   */
+  dependencies?: ReadonlyArray<Dependency>;
+
+  /**
+   * Any Alpine packages needed at runtime.
+   */
+  runtimeDeps?: ReadonlyArray<string>;
+
+  /**
+   * Commands to run after all builds have finished - use to inject configuration
+   * or otherwise customize the container's environment.
+   */
+  postBuildCommands?: ReadonlyArray<string | ReadonlyArray<string>>;
+
+  /**
+   * The `USER` directive to inject at the end of the build.
+   */
+  user?: string;
+}
+
+const runDepsCommand = `runDeps="$(\\
+    scanelf --needed --nobanner --format '%n#p' --recursive /usr/local \\
+      | tr ',' '\\n' \\
+      | sort -u \\
+      | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \\
+  )"`;
+
+/**
+ * Helper type used in `createPHPDockerfile()`.
+ */
+interface PeclPackage {
+  /**
+   * The name of the PECL package.
+   */
+  name: string;
+  /**
+   * Whether or not to enable it by default in the image.
+   */
+  enabled: boolean;
+}
+
+function createPHPDockerfile({
```

Comment:
It looks like one path forward for this could be breaking it down by the step-level inline comments you've got in place already. Some will be relatively simple and straightforward, but the improved readability and testability would be helpful.